### ISSUE TITLE
Presto is not able to connect to hive metastore when hadoop.rpc.protection=privacy

### ISSUE THREAD

**Issue by scorta-d** [prestodb/presto#8604]
If hadoop.rpc.proctecion=privacy then presto fails to connect to hive metastore
```
<property>
  <name>hadoop.rpc.protection</name>
  <value>privacy</value>
</property>

```

The error message is `Could not connect to meta store using any of the URIs provided. M
ost recent failure: org.apache.thrift.transport.TTransportException: No common protection layer between client and server`

It is possible to patch presto code in order to fix this:
In KerberosHiveMetastoreAuthentication.java  file the line 

```
Sasl.QOP, "auth",
```

should be replaced with 

```
Sasl.QOP, "auth-conf",
```

The standard hive mestastore client works out of the box when core-site.xml and hive-site.xml are added to HiveConf as resources.

**Comment by akshatnair** [comment]
We need to make this a configuration property for Hive connector.

**Comment by kokosing** [comment]
Fixed with #11013
